rpc/eth_api.go

```diff
@@ -341,8 +340,19 @@ type CallArgs struct {
 
 // Call performs a raw contract call.
 func (e *PublicEthAPI) Call(args CallArgs, blockNr rpc.BlockNumber, overrides *map[common.Address]account) (hexutil.Bytes, error) {
-	result, err := e.doCall(args, blockNr, vm.Config{}, big.NewInt(emint.DefaultRPCGasLimit))
-	return (hexutil.Bytes)(result), err
+	result, err := e.doCall(args, blockNr, big.NewInt(emint.DefaultRPCGasLimit))
+	if err != nil {
+		return []byte{}, err
+	}
+	var simResult sdk.Result
+	if err = e.cliCtx.Codec.UnmarshalBinaryLengthPrefixed(result, &simResult); err != nil {
+		return []byte{}, err
+	}
+	_, _, ret, err := types.DecodeReturnData(simResult.Data)
+	if err != nil {
```

Comment:
This error check is not necessary, if the endpoint returns an error (as it does below) the return value would be ignored anyway.